From 7d8180abcc5b5c8bbad42bfae8eefa931c3bcdfa Mon Sep 17 00:00:00 2001
From: Will Thompson <wjt@endlessm.com>
Date: Tue, 26 May 2020 22:27:12 +0100
Subject: [PATCH] gtk: Use reversed-domain icon name throughout

In the Flatpak, we rename the icon from 'transmission' to
'com.transmissionbt.Transmission' to map the app ID (as required by
Flatpak's rules around exporting files).

Adjust the GTK default window icon and libindicator icon to match.

Fixes https://github.com/flathub/com.transmissionbt.Transmission/issues/18
---
 gtk/main.c    | 2 +-
 gtk/tr-icon.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/gtk/main.c b/gtk/main.c
index 8bc526204..45d0ec3de 100644
--- a/gtk/main.c
+++ b/gtk/main.c
@@ -659,7 +659,7 @@ int main(int argc, char** argv)
         return 0;
     }
 
-    gtk_window_set_default_icon_name(MY_CONFIG_NAME);
+    gtk_window_set_default_icon_name("com.transmissionbt.Transmission");
 
     /* init the unit formatters */
     tr_formatter_mem_init(mem_K, _(mem_K_str), _(mem_M_str), _(mem_G_str), _(mem_T_str));
diff --git a/gtk/tr-icon.c b/gtk/tr-icon.c
index 15d4e22eb..81870a5a4 100644
--- a/gtk/tr-icon.c
+++ b/gtk/tr-icon.c
@@ -19,7 +19,7 @@
 
 static TR_DEFINE_QUARK(tr_core, core)
 
-#define ICON_NAME "transmission"
+#define ICON_NAME "com.transmissionbt.Transmission"
 
 #ifdef HAVE_LIBAPPINDICATOR
 
-- 
2.20.1

